src/librustc_typeck/astconv.rs

```diff
@@ -306,6 +306,21 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             };
         }
 
+        for (param, binding) in def.params.iter().zip(args.bindings.iter()) {
```

Comment:
This should be iterating over `args.args`, yes. Also, it's probably better to move this after the calls to `check_kind_counts`, so we get a "mismatched number of arguments" error first.

Reply:
I'm thinking `def.params.iter()` is wrong too.
```rust
fn new_foo(f: Foo) {}
```
still causes iteration over `T` and as far as I can tell `param`'s `DefId`s are not being marked as sable what ever I do.

Any thoughts?

```rust
pub struct Foo<#[unstable(feature = "unstable_generic_feature", issue = "0")] T = usize> {
    _f: T,
}
```

Reply:
I am trying moving it to `create_substs_for_ast_path()`